### PR TITLE
Fix buy button adding to minicart when you have more than one SKU

### PR DESCRIPTION
**Pull request by Klynger**
#### What is the purpose of this pull request?
When you click on the `BuyButton` and the product has more than one SKU you should go to the Product page instead of add to the mini-cart

#### What problem is this solving?
`BuyButton` is adding the first SKU of the product when you click on it and you have more than one SKU

#### How should this be manually tested?
Go to the [workspace](https://summarybuybutton--storecomponents.myvtex.com/classic?_q=classic&map=ft) and click on the `BuyButton`. You should go to the product page instead of adding to the mini-cart.

#### Types of changes
- [X] Bugfix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js [hunk 1]
```diff
@@ -14,6 +16,18 @@ import displayButtonTypes, {
...
+function checkSKUAvailability(seller) {
+  return path(['commertialOffer', 'AvailableQuantity'], seller) > 0
+}
+
+function checkAvailableSKUsInSellers(item) {
+  return item.sellers && item.sellers.filter(checkSKUAvailability).length > 0
+}
+
+function shouldAddToCart(items) {
+  return items.filter(checkAvailableSKUsInSellers).length === 1
```
Comment (klzns) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: I think the logic here is even simpler.

Just `items.length > 1`

Comment (Klynger) [reply react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: But if it is bigger than 1 you have more than one SKU, so you can't add a random SKU to the cart

Comment (lbebber) [reply react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: does it add to cart if the item has, say, 3 SKUs, but only 1 available? if so, I don't think this behaviour is desired.

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js [hunk 2]
```diff
@@ -59,18 +73,36 @@ const ProductSummaryBuyButton = ({
     selectedQuantity,
   })
 
+  const { items = [] } = product
+  const haveMoreSKUs = shouldAddToCart(items)
```
Comment (lbebber) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: `hasMultipleSKUs`

Comment (victorhmp) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]:
```suggestion
  const hasMoreSKUs = shouldAddToCart(items)
```

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js [hunk 3]
```diff
@@ -59,18 +73,36 @@ const ProductSummaryBuyButton = ({
...
+              params={{
+                slug: product && product.linkText,
+                id: product && product.productId,
+              }}
+            >
+              <Button>
+                <IOMessage id={buyButtonText} />
+              </Button>
+            </Link>
+          )}
```
Comment (victorhmp) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: This logic might be reversed

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js [hunk 4]
```diff
@@ -14,6 +16,18 @@ import displayButtonTypes, {
...
 
+function checkSKUAvailability(seller) {
+  return path(['commertialOffer', 'AvailableQuantity'], seller) > 0
+}
+
+function checkAvailableSKUsInSellers(item) {
+  return item.sellers && item.sellers.filter(checkSKUAvailability).length > 0
+}
+
+function hasOneSKU(items) {
```
Comment (iaronaraujo) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]:
```suggestion
function hasOnlyOneSKU(items) {
```

The `Only` in the name makes it clearer that this not true when there are more than one SKU

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js (hunk 1 above)
Comment (Klynger) [reply react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: It does, the function `checkSKUAvailability` does this check.

Comment (lbebber) [reply react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: (talking about this behavior over DM)

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js [hunk 5]
```diff
@@ -14,6 +16,10 @@ import displayButtonTypes, {
 
 import productSummary from '../../productSummary.css'
 
+const ALWAYS_GO_TO_PRODUCT = 'alwaysGoToProduct'
+const ADD_TO_CART = 'addToCart'
+const BUY_BUTTOM_BEHAVIOR_OPTIONS = [ALWAYS_GO_TO_PRODUCT, ADD_TO_CART]
```
Comment (victorhmp) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]:
```suggestion
const BUY_BUTTON_BEHAVIOR_OPTIONS = [ALWAYS_GO_TO_PRODUCT, ADD_TO_CART]
```

File: react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js [hunk 6]
```diff
@@ -14,6 +16,10 @@ import displayButtonTypes, {
 
 import productSummary from '../../productSummary.css'
 
+const ALWAYS_GO_TO_PRODUCT = 'alwaysGoToProduct'
+const ADD_TO_CART = 'addToCart'
```
Comment (lbebber) [inline react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: (pensando se tem um nome melhor pra esse modo; ele nao vai necessariamente adicionar ao carrinho sempre, ele vai adicionar só se nao tiver variaçoes né?)

Comment (lbebber) [reply react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js]: acho q pode ser só `default` mesmo, e explicar isso na documentaçao 🤷‍♂

**Pull request merged**